MapboxVision/Services/Recording/DocumentsLocation.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+enum DocumentsLocation: Equatable {
+    case currentRecording
+    case recordings(SyncRegion)
+    case cache
+    case custom
+
+    private var directoryName: String {
+        switch self {
+        case .currentRecording:
+            return "Current"
+        case .cache:
+            return "Cache"
+        case let .recordings(region):
+            return "Recording_\(region.rawValue)"
+        case .custom:
+            return ""
+        }
+    }
+
+    var path: String {
+        let searchDirectory: FileManager.SearchPathDirectory?
+
+        switch self {
+        case .currentRecording, .cache, .recordings:
+            searchDirectory = .cachesDirectory
+        case .custom:
+            searchDirectory = nil
+        }
+
+        var basePath: String
+
+        if
+            let searchDirectory = searchDirectory,
+            let baseDirectory = NSSearchPathForDirectoriesInDomains(searchDirectory, .userDomainMask, true).first
+        {
```

Comment:
Probably we can simplify `path` implementation a bit then?

```swift
    var path: String {
        var basePath: String

        switch self {
        case .currentRecording, .cache, .recordings:
            if let cachesDirPath = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true).first {
                basePath = cachesDirPath
            } else {
                basePath = NSTemporaryDirectory()
            }
        case .custom:
            basePath = NSTemporaryDirectory()
        }

        if let bundleIdentifier = Bundle(for: BundleToken.self).bundleIdentifier {
            basePath = basePath.appendingPathComponent(bundleIdentifier, isDirectory: true)
        }

        return basePath.appendingPathComponent(directoryName, isDirectory: true)
    }
```

Reply:
```swift
    var path: String {
        var basePath: String = NSTemporaryDirectory()

        if
            self != .custom,
            let cachesPath = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true).first
        {
            basePath = cachesPath
        }

        if let bundleIdentifier = Bundle(for: BundleToken.self).bundleIdentifier {
            basePath = basePath.appendingPathComponent(bundleIdentifier, isDirectory: true)
        }

        return basePath.appendingPathComponent(directoryName, isDirectory: true)
    }
```
?

Reply:
Looks good!